PluginGenerator/PluginBuilder.cs

```diff
@@ -17,14 +17,34 @@ public class PluginBuilder
     {
         public const string SONARQUBE_API_VERSION = "4.5.2";
 
+        /// <summary>
+        /// The name of the package in which the plugin will be created
+        /// </summary>
+        private const string CorePluginPackageNameTemplate = "org.sonarqube.plugin.sdk.[PLUGIN_KEY]";
+
+        /// <summary>
+        /// The name of the class that SonarQube should load to discover the
+        /// extensions offered by the plugin
+        /// </summary>
+        private const string CorePluginEntryClassTemplate = "[PLUGIN_PACKGE].Plugin";
+
+        /// <summary>
+        /// Token that should be replaced with a list of the classes that exposed by
+        /// the plugin as a extensions.
+        /// Sample value: "org.sonarqube.plugin.sdk.MyRulesDefinition.class, myorg.otherExtension.class"
+        /// </summary>
+        private const string ExtensionListToken = "[CORE_EXTENSION_CLASS_LIST]";
+
         private readonly IJdkWrapper jdkWrapper;
         private readonly ILogger logger;
 
         private readonly ISet<string> sourceFiles;
         private readonly ISet<string> referencedJars;
+        private readonly ISet<string> extensionClasses;
 
         private readonly IDictionary<string, string> properties;
         private readonly IDictionary<string, string> fileToRelativePathMap;
+        private readonly IDictionary<string, string> sourceFileReplacements;
```

Comment:
The responsibility for replacing tokens in source files has moved to the core builder.